tests/benchmark/unit/src/test_c_allocators.cpp

```diff
@@ -0,0 +1,174 @@
+#include <iostream>
+
+#include <gtest/gtest.h>
+
+#include "memory_management.hpp"
+#include "test_config.hpp"
+#include "test_types.hpp"
+
+TEST(CAllocators, MallocReallocFree)
```

Comment:
i suggest splitting this into several test of CAllocators family

Reply:
I bundled those for a reason: realloc is usually used with malloc (or calloc) followed by free. It can in fact work with free alone if the ptr passed it is null (it uses malloc internally). So Reallon, malloc or calloc without a free will leak. We can configure the test to detect how much was leaked. But I don't think that's a reasonable way of thinking. Any thought?